### PR TITLE
Add further extension to the Azure VM table to support security feature query. closes #46

### LINKED ISSUE #46: Add further extension to the Azure VM table to support security feature query
**Is your feature request related to a problem? Please describe.**
Would like to have following details associated with the vm table such as --

How to know of the Vm is associated with any endpoint protection e.g. az vm show -g MyResourceGroup -n MyVm -d
Also provide support for the vm extension such as : az vm extension list -g MyResourceGroup --vm-name MyVm
Describe the solution you'd like

This is to achieve some of the CIS steps ( such as 7.4, 7.5, 7.6)

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by bigdatasourav**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
Add passing integration test logs here
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
> select jsonb_pretty(extensions) as extensions from azure_compute_virtual_machine;
+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| extensions                                                                                                                                                                              |
+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| [                                                                                                                                                                                       |
|     {                                                                                                                                                                                   |
|         "Id": "/subscriptions/d7245080-b4ae-4fe5-b6fa-2e71b3dae6c8/resourceGroups/TURBOT_RG/providers/Microsoft.Compute/virtualMachines/testExtension/extensions/NvidiaGpuDriverLinux", |
|         "Name": "NvidiaGpuDriverLinux",                                                                                                                                                 |
|         "Type": "Microsoft.Compute/virtualMachines/extensions",                                                                                                                         |
|         "tags": null,                                                                                                                                                                   |
|         "Location": "westus",                                                                                                                                                           |
|         "Settings": {                                                                                                                                                                   |
|         },                                                                                                                                                                              |
|         "Publisher": "Microsoft.HpcCompute",                                                                                                                                            |
|         "InstanceView": null,                                                                                                                                                           |
|         "ForceUpdateTag": null,                                                                                                                                                         |
|         "ProtectedSettings": null,                                                                                                                                                      |
|         "ProvisioningState": "Succeeded",                                                                                                                                               |
|         "TypeHandlerVersion": "1.2",                                                                                                                                                    |
|         "EnableAutomaticUpgrade": null,                                                                                                                                                 |
|         "AutoUpgradeMinorVersion": true                                                                                                                                                 |
|     }                                                                                                                                                                                   |
| ]                                                                                                                                                                                       |
+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+

```
</details>

### CODE REVIEW

File: azure/table_azure_compute_virtual_machine.go [hunk 1]
```diff
@@ -267,7 +267,13 @@ func tableAzureComputeVirtualMachine(_ context.Context) *plugin.Table {
 				Type:        proto.ColumnType_JSON,
 				Hydrate:     getAzureComputeVirtualMachineStatuses,
 			},
-
+			{
+				Name:        "extensions",
+				Description: "Specifies the VM Extensions details",
```
Comment (Subhajit97) [inline azure/table_azure_compute_virtual_machine.go:272]:
```suggestion
				Description: "Specifies the details of VM Extensions.",
```

File: azure/table_azure_compute_virtual_machine.go [hunk 2]
```diff
@@ -474,6 +480,48 @@ func getNicPublicIP(ctx context.Context, session *Session, id string) (network.P
...
+		extensionMap["Publisher"] = extension.VirtualMachineExtensionProperties.Publisher
+		extensionMap["TypeHandlerVersion"] = extension.VirtualMachineExtensionProperties.TypeHandlerVersion
+		extensionMap["AutoUpgradeMinorVersion"] = extension.VirtualMachineExtensionProperties.AutoUpgradeMinorVersion
+		extensionMap["EnableAutomaticUpgrade"] = extension.VirtualMachineExtensionProperties.EnableAutomaticUpgrade
+		extensionMap["ForceUpdateTag"] = extension.VirtualMachineExtensionProperties.ForceUpdateTag
+		extensionMap["Settings"] = extension.VirtualMachineExtensionProperties.Settings
+		extensionMap["ProtectedSettings"] = extension.VirtualMachineExtensionProperties.ProtectedSettings
+		extensionMap["ProvisioningState"] = extension.VirtualMachineExtensionProperties.ProvisioningState
+		extensionMap["InstanceView"] = extension.VirtualMachineExtensionProperties.InstanceView
+		extensionMap["tags"] = extension.Tags
```
Comment (Subhajit97) [inline azure/table_azure_compute_virtual_machine.go:519]:
```suggestion
		extensionMap["VirtualMachineExtensionProperties"] = extension.VirtualMachineExtensionProperties
```

File: azure/table_azure_compute_virtual_machine.go [hunk 3]
```diff
@@ -474,6 +480,48 @@ func getNicPublicIP(ctx context.Context, session *Session, id string) (network.P
...
+	subscriptionID := session.SubscriptionID
+	client := compute.NewVirtualMachineExtensionsClient(subscriptionID)
+	client.Authorizer = session.Authorizer
+
+	op, err := client.List(context.Background(), resourceGroupName, *virtualMachine.Name, "")
+	if err != nil {
+		return nil, err
+	}
+
+	var extensions []map[string]interface{}
```
Comment (Subhajit97) [inline azure/table_azure_compute_virtual_machine.go:502]: Add a comment to describe the reason for handling the following case

**Review by Subhajit97 — CHANGES_REQUESTED** [review]
@bigdatasourav There are few changes required. Please see comments, thanks!

File: azure/table_azure_compute_virtual_machine.go (hunk 2 above)
Comment (bigdatasourav) [reply azure/table_azure_compute_virtual_machine.go:519]: as discussed, properties need to fetch in that way

File: azure/table_azure_compute_virtual_machine.go [hunk 4]
```diff
@@ -474,6 +480,50 @@ func getNicPublicIP(ctx context.Context, session *Session, id string) (network.P
...
+	client := compute.NewVirtualMachineExtensionsClient(subscriptionID)
+	client.Authorizer = session.Authorizer
+
+	op, err := client.List(context.Background(), resourceGroupName, *virtualMachine.Name, "")
+	if err != nil {
+		return nil, err
+	}
+
+	// If we return the API response directly, the output only gives the contents of VirtualMachineExtensionsListResult
+
```
Comment (Subhajit97) [inline azure/table_azure_compute_virtual_machine.go:503]:
```suggestion
```

**Review by Subhajit97 — APPROVED** [review]
LGTM

File: azure/table_azure_compute_virtual_machine.go [hunk 5]
```diff
@@ -474,6 +480,49 @@ func getNicPublicIP(ctx context.Context, session *Session, id string) (network.P
...
+		extensionMap["Publisher"] = extension.VirtualMachineExtensionProperties.Publisher
+		extensionMap["TypeHandlerVersion"] = extension.VirtualMachineExtensionProperties.TypeHandlerVersion
+		extensionMap["AutoUpgradeMinorVersion"] = extension.VirtualMachineExtensionProperties.AutoUpgradeMinorVersion
+		extensionMap["EnableAutomaticUpgrade"] = extension.VirtualMachineExtensionProperties.EnableAutomaticUpgrade
+		extensionMap["ForceUpdateTag"] = extension.VirtualMachineExtensionProperties.ForceUpdateTag
+		extensionMap["Settings"] = extension.VirtualMachineExtensionProperties.Settings
+		extensionMap["ProtectedSettings"] = extension.VirtualMachineExtensionProperties.ProtectedSettings
+		extensionMap["ProvisioningState"] = extension.VirtualMachineExtensionProperties.ProvisioningState
+		extensionMap["InstanceView"] = extension.VirtualMachineExtensionProperties.InstanceView
+		extensionMap["tags"] = extension.Tags
```
Comment (cbruno10) [inline azure/table_azure_compute_virtual_machine.go:520]: Why is this lowercase `tags`, should it be `Tags` (seems inconsistent with the others)?

**Review by cbruno10 — CHANGES_REQUESTED** [review]
Please see comment, thanks!

File: azure/table_azure_compute_virtual_machine.go (hunk 5 above)
Comment (bigdatasourav) [reply azure/table_azure_compute_virtual_machine.go]: updated

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**